### ISSUE TITLE
Preparing the frontend (?) create a target directory; files should exist in build

### ISSUE THREAD

**Issue by christoph-frick** [vaadin/vaadin-gradle-plugin#46]
**Desktop (please complete the following information):**

 - OS: Linux
 - JDK version: Zulu 8
 - Gradle version: 6.0.1
 - Vaadin Plugin version: 0.5

**Describe the bug**

Probably `vaadinPrepareFrontend` creates a `target` directory in the
root.

**To Reproduce**

Given a simple project like
https://github.com/christoph-frick/try-vaadin14-gradle/blob/master/build.gradle
building or running the project (assumption: the task
`vaadinPrepareFrontend` is the culprit), the project contains a `target`
directory afterwards, which usually is associated with a maven build:

```sh
$ find target
target
target/frontend
target/frontend/generated-flow-imports.js
target/frontend/package.json
target/frontend/generated-flow-imports-fallback.js
```

**Expected behavior**

All build artifacts should reside somewhere inside the `build`
directory.

**Screenshots**

n/a

**Additional context**

n/a

**Comment by mvysny** [comment]
I completely agree. Unfortunately there's a catch as hinted at https://github.com/vaadin/vaadin-gradle-plugin/blob/master/src/main/kotlin/com/vaadin/gradle/VaadinFlowPluginExtension.kt#L71 :

> In order to have stuff in build/ instead of target/ we need Flow 2.2 or higher.

Flow 2.2 means Vaadin 14.2+; that would mean that the plugin would not support apps using Vaadin 14.1.x (since the VaadinServlet in development mode expects stuff to reside in `target/`).

However, I propose to leave this report open, since at some point it should be good to follow Gradle convention and generate files in build/ .

**Comment by lightoze** [comment]
Perhaps this can be now fixed in 17 branch?

**Comment by mvysny** [comment]
Definitely! Let me try.

**Comment by mvysny** [comment]
Hmm, I'm not sure it's possible yet. The `target/` path is hard-coded in the `FrontendUtils.TARGET` constant, which is also used for example by `DEFAULT_GENERATED_DIR`. The value is then read by `DevModeInitializer` class and `NodeTasks` class, and both use the same approach: the value may only be overridden via a Java System property `FrontendUtils.PARAM_GENERATED_DIR` (`vaadin.frontend.generated.folder`).

Therefore, it's not enough to only configure the value in the Gradle plugin - the value must probably be also configured in runtime in the Vaadin app itself. Please let me verify this finding; if it is indeed so, I will open a feature request for Flow to move this property to e.g. `flow-build-info.json`.

**Comment by mvysny** [comment]
Even if you try to change the value of `generatedFolder` to e.g. `generatedFolder = File(project.projectDir, "build/frontend")`, Vaadin will still generate a bunch of files into `target/` folder:

```
├── flow-frontend
│   ├── form
│   │   └── package.json
│   └── package.json
├── index.html
└── index.ts
```

**Comment by mvysny** [comment]
Let me open a bug report at https://github.com/vaadin/flow/issues to start a discussion in order to possibly fix this.

**Comment by mvysny** [comment]
Blocked by https://github.com/vaadin/flow/issues/8953

**Comment by mvysny** [comment]
Blocked by https://github.com/vaadin/flow/issues/10335
